provisioner/terraform/tfparse/tfparse.go

```diff
@@ -316,14 +316,28 @@ func (p *Parser) CoderParameterDefaults(ctx context.Context) (map[string]string,
 			}
 
 			if _, ok := resContent.Attributes["default"]; !ok {
+				p.logger.Warn(ctx, "coder_parameter data source does not have a default value", slog.F("name", dataResource.Name))
 				defaultsM[dataResource.Name] = ""
 			} else {
 				expr := resContent.Attributes["default"].Expr
 				value, err := previewFileContent(expr.Range())
 				if err != nil {
 					return nil, xerrors.Errorf("can't preview the resource file: %v", err)
 				}
-				defaultsM[dataResource.Name] = strings.Trim(value, `"`)
+				// Issue #15795: the "default" value could also be an expression we need
+				// to evaluate.
+				// TODO: should we support coder_parameter default values that reference other coder_parameter data sources?
```

Comment:
Has there been an ask? If not, we can probably defer it until such a time. It's best to be careful how much we expand the scope of these expressions.

Reply:
I'm not aware of any ask for this just yet. I'd much prefer to defer it as it opens up several cans of worms.